src/test/java/nl/tudelft/ewi/devhub/webtests/views/DeliveryReviewView.java

```diff
@@ -0,0 +1,196 @@
+            final List<WebElement> ddElements = getDriver().findElements(By.cssSelector("blockquote dl dd"));
+            if (ddElements.size() == 2) {
+                ddElements.get(0).getText();
+            } else if (getDriver().findElement(By.cssSelector("blockquote dl dt")).getText().equalsIgnoreCase("Grade")) {
+                return Double.parseDouble(ddElements.get(0).getText());
+            }
+
+            return 0D;
+        }
+
+        @Nullable
+        public String getCommentary() {
+            final List<WebElement> ddElements = getDriver().findElements(By.cssSelector("blockquote dl dd"));
+            if (ddElements.size() == 2) {
+                ddElements.get(1).getText();
+            } else if (getDriver().findElement(By.cssSelector("blockquote dl dt")).getText().equalsIgnoreCase("Remarks")) {
+                return ddElements.get(0).getText();
+            }
+
+            return null;
+        }
+
+
+        @Nullable
+        public Date getReviewTime() throws ParseException {
+            final Optional<String> footer = getFooter();
+            if (!footer.isPresent()) {
+                return null;
+            }
+
+            return new SimpleDateFormat("EEEE dd MMMM yyyy k:m", Locale.US).parse(footer.get().split(" on ")[1].trim());
+        }
+
+        private Optional<String> getFooter() {
+            return getDriver().findElements(By.cssSelector("footer.small")).stream().findAny().map(WebElement::getText);
+        }
+    }
+
+
+    public class DeliveryForm {
+        private final WebElement anchor;
+        private final By STATE_SELECTOR = By.cssSelector("select#state");
+        private final By GRADE_SELECTOR = By.cssSelector("input#grade");
+        private final By COMMENTARY_SELECTOR = By.cssSelector("textarea#commentary");
+
+        DeliveryForm(WebElement anchor) {
+            this.anchor = anchor;
+        }
+
+        public State getSelectedState() {
+            final WebElement element = getDriver().findElement(STATE_SELECTOR);
+            Select select = new Select(element);
+
+            return State.valueOf(select.getFirstSelectedOption().getText().toUpperCase());
+        }
+
+        /**
+         * options:
+         * - Submitted
+         * - Rejected
+         * - Approved
+         * - Disapproved
+         *
+         * @param newState
+         * @throws IllegalArgumentException
+         */
+        public void setState(String newState) throws IllegalArgumentException {
+            invariant();
+            final WebElement element = getDriver().findElement(STATE_SELECTOR);
+            final Select select = new Select(element);
+
+            final Optional<WebElement> optionalNewStateElement = select.getOptions()
+                    .stream()
+                    .filter(webElement -> webElement.getText().equals(newState)).findAny();
+
+            if (optionalNewStateElement.isPresent()) {
+                optionalNewStateElement.get().click();
```

Comment:
The `get` will throw an `NoSuchElementException` if not present. As you expect the element to exist, and want to throw an exception when it doesn't anyway, you could just write:

```java
select.getOptions()
  .stream()
  .filter(webElement -> webElement.getText().equals(newState)).findAny()
  .get().click();
```